From ace2a6608e4b8948baee5b9c89ba62c39cbf30b8 Mon Sep 17 00:00:00 2001
From: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>
Date: Sat, 4 Apr 2020 21:57:03 -0700
Subject: [PATCH] make-initrd: cross install tweaks

Signed-off-by: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@xilinx.com>
---
 initrd/make-initrd | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/initrd/make-initrd b/initrd/make-initrd
index c087137..4f23cb2 100755
--- a/initrd/make-initrd
+++ b/initrd/make-initrd
@@ -16,8 +16,21 @@ mkdir -p $tmpdir/sys
 mkdir -p $tmpdir/lib
 mkdir -p $tmpdir/var
 mkdir -p $tmpdir/mnt
-cp `which busybox` $tmpdir/bin
-$tmpdir/bin/busybox --install $tmpdir/bin
+
+if [ -z "$BUSYBOX" ]; then
+    BUSYBOX=`which busybox`
+fi
+cp $BUSYBOX $tmpdir/bin
+if [ -n "$CROSS_COMPILE" ]; then
+    echo "cross compiling, busybox --install emulation"
+    if [ -n "$QEMU_USER" ]; then
+        $QEMU_USER $tmpdir/bin/busybox --install $tmpdir/bin
+    else
+        echo "QEMU_USER is not defined, no binary symlinks will be available"
+    fi
+else
+    $tmpdir/bin/busybox --install $tmpdir/bin
+fi
 
 cp $init $tmpdir/init
 chmod +x $tmpdir/init
-- 
2.7.4

